server/sms/gateway.go

```diff
@@ -0,0 +1,242 @@
+package gateway
+
+import (
+	"context"
+	"encoding/binary"
+	log "github.com/Sirupsen/logrus"
+	"github.com/smancke/guble/protocol"
+	"github.com/smancke/guble/server/connector"
+	"github.com/smancke/guble/server/router"
+	routerimport "github.com/smancke/guble/server/router"
+	"github.com/smancke/guble/server/store"
+	"strconv"
+)
+
+const SMS_SCHEMA = "sms_notifications"
+const SMS_DEFAULT_TOPIC = "sms"
+
+type Sender interface {
+	Send(*protocol.Message) error
+}
+
+type Config struct {
+	Enabled   *bool
+	APIKey    *string
+	APISecret *string
+	Workers   *int
+	SMSTopic  *string
+
+	Name      string
+	Schema    string
+}
+
+type gateway struct {
```

Comment:
Would be ideal to name this Connector and the gateway to be the sender. The name gateway comes from SMS Gateway and it should be an interface that actually takes the destination and body and send it. And this library should be externally configured and not together with the SMS. (eg.: The api key and api secret is something specific to nexmo, may not be specific to a different provider)